ts/Stock/StockTools/StockToolbar.ts

```diff
@@ -707,11 +726,67 @@ class Toolbar {
     }
 
     /**
-     * Redraw, GUI requires to verify if the navigation should be visible.
+     * Redraws the toolbar based on the current state of the options.
      * @private
      */
     public redraw(): void {
-        this.showHideNavigatorion();
+        if (this.options.enabled !== this.guiEnabled) {
+            if (this.options.enabled === false) {
+                this.destroy();
+                this.guiEnabled = false;
+                this.visible = false;
+            }
+
+            if (this.options.enabled === true) {
+                this.createContainer();
+                this.createButtons();
+            }
+
+            this.guiEnabled = this.options.enabled;
+        } else {
+            if (!this.guiEnabled) {
+                return;
+            }
+
+            if (this.options.className !== this.guiClassName) {
+                if (this.guiClassName) {
+                    this.wrapper.classList.remove(this.guiClassName);
+                }
+                if (this.options.className) {
+                    this.wrapper.classList.add(this.options.className);
+                }
+                this.guiClassName = this.options.className;
+            }
+
+            if (this.options.toolbarClassName !== this.toolbarClassName) {
+                if (this.toolbarClassName) {
+                    this.toolbar.classList.remove(this.toolbarClassName);
+                }
+                if (this.options.toolbarClassName) {
+                    this.toolbar.classList.add(this.options.toolbarClassName);
+                }
+                this.toolbarClassName = this.options.toolbarClassName;
+            }
+
+            if (
+                !shallowArraysEqual(this.options.buttons, this.buttonList) ||
+                this.isDirty
+            ) {
+                this.toolbar.innerHTML = AST.emptyHTML;
+                this.createButtons();
+            }
+
+            if (this.options.enabled === false) {
+                this.destroy();
+            }
+
+            if (defined(this.options.visible)) {
+                this.visible = this.options.visible;
+            }
```

Comment:
There are a lot of if statements here. Would you mind splitting that into separate methods to improve code readability?